tests/internal/helpers.js

```diff
@@ -50,7 +51,16 @@ import { describe, expect } from 'https://jslib.k6.io/k6chaijs/4.3.4.1/index.js'
  * ```
  */
 export async function asyncDescribe(description, callback) {
-    await callback((what) => {
-        return expect(what, description)
-    })
+    try {
+        await callback((what) => {
+            return expect(what, description)
+        })
+    } catch (error) {
+        if (error.name !== 'AssertionError') {
+            console.error(`FAIL [${description}]`, error.stack);
+            check(error, {[description]: () => false})
+        } else {
+            console.error(`FAIL [${description}]`, error.message);
+        }
+    }
```

Comment:
Adding this bit to provide better error messages. While testing I noticed that we weren't really adding a failed check when the test failed without any failed assertions.